### PR TITLE
Protocol count validation for Macaques

### PR DESCRIPTION
**Pull request by labkey-martyp**
#### Rationale
This PR creates a new WNPRC specific verifyProtocolCounts function similar to the EHR version but accounting for Macaque species validation. The EHR assignment trigger is unregistered in favor of a WNPRC trigger calling the new verifyProtocolCounts function.

#### Related Pull Requests
* Replacing this PR https://github.com/LabKey/ehrModules/pull/791
* https://github.com/LabKey/ehrModules/pull/795

#### Changes
* Add the verifyProtocolCounts function from PR referenced above
* Unregister EHR assignment trigger
* Update WNPRC assignment trigger to use WNPRC verifyProtocolCounts
* Copy protocolTotalAnimalsBySpecies.sql from the above PR

### CODE REVIEW

File: WNPRC_EHR/test/src/org/labkey/test/tests/wnprc_ehr/WNPRC_EHRTest.java [hunk 1]
```diff
@@ -274,8 +280,7 @@ public static void doSetup() throws Exception
         initTest.setupAnimalRequests();
 
         initTest.checkUpdateProgramIncomeAccount();
-
-        initTest.deathNotificationSetup();
+        //initTest.deathNotificationSetup();
```
Comment (labkey-jeckels) [inline WNPRC_EHR/test/src/org/labkey/test/tests/wnprc_ehr/WNPRC_EHRTest.java]: Should this get re-enabled?

File: WNPRC_EHR/src/org/labkey/wnprc_ehr/TriggerScriptHelper.java [hunk 2]
```diff
@@ -2413,4 +2417,110 @@ public Boolean checkIfUserIsWaterAdmin(int userId, int project, User currentUser
...
+        if (protocol == null)
+        {
+            return "Unable to find protocol associated with project: " + project;
+        }
+
+        //find the total animals previously used by this protocols/species
+        TableInfo ti = QueryService.get().getUserSchema(user, container, "study").getTable("protocolTotalAnimalsBySpecies");
+        String animalSpecies = ar.getSpecies();
+        SimpleFilter filter;
+        if (("Rhesus".equals(animalSpecies) || "Cynomolgus".equals(animalSpecies)))
```
Comment (labkey-jeckels) [inline WNPRC_EHR/src/org/labkey/wnprc_ehr/TriggerScriptHelper.java]: Seems like it's time to introduce constants for the species names

**Review by labkey-jeckels — APPROVED** [review]
Looks OK to me but will defer to Chad on the business logic review.

File: WNPRC_EHR/test/src/org/labkey/test/tests/wnprc_ehr/WNPRC_EHRTest.java (hunk 1 above)
Comment (csebranek) [reply WNPRC_EHR/test/src/org/labkey/test/tests/wnprc_ehr/WNPRC_EHRTest.java]: Yes

**Review by csebranek — APPROVED** [review]

**Pull request merged**